### PR TITLE
Fixes #11308: make a dsc user_technique

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/11308

### CODE REVIEW

File: techniques/systemSettings/userManagement/userManagement/8.0/userManagement.st [hunk 1]
```diff
@@ -0,0 +1,526 @@
+#####################################################################################
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/8.0/userManagement.st]: Same file as in the previous version

**Comment by Fdall** [comment]
Need to change user_present to the new name according to https://www.rudder-project.org/redmine/issues/11332

**Comment by Fdall** [comment]
Commit modified

File: techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,191 @@
+<!--
+Copyright 2016 Normation SAS
```
Comment (amousset) [inline techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml]: 2017

File: techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<TECHNIQUE name="Users">
+  <DESCRIPTION>This technique manages the target host(s) users.
+  It will ensure that the defined users are present on the system.</DESCRIPTION>
+  <MULTIINSTANCE>true</MULTIINSTANCE>
```
Comment (amousset) [inline techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml]: We should not remove compatible OS information.

File: techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        </ITEM>
+        <CONSTRAINT>
+          <DEFAULT>everytime</DEFAULT>
+        </CONSTRAINT>
+      </SELECT1>
+      <SECTION name="Password" multivalued="false" component="true" componentKey="USERGROUP_USER_LOGIN">
+        <INPUT>
+          <NAME>USERGROUP_USER_PASSWORD</NAME>
+          <DESCRIPTION>Password for this account</DESCRIPTION>
+          <LONGDESCRIPTION>Windows agent only support "clear text" entries at the moment</LONGDESCRIPTION>
```
Comment (amousset) [inline techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml]: supports

File: techniques/systemSettings/userManagement/userManagement/8.0/userManagement.ps1.st [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    &USERGROUP_USER_PASSWORD:{password |
+    "&password&" };separator=","& )
+
+    $password_policies = @(
+    &USERGROUP_USER_PASSWORD_POLICY:{password_policy |
+    "&password_policy&" };separator=","& )
+
+    $present        = "add"
+    $absent         = "remove"
+#To REMOVE in the future
```
Comment (amousset) [inline techniques/systemSettings/userManagement/userManagement/8.0/userManagement.ps1.st]: why?

File: techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+      <DESCRIPTION>User ID (enforced at user creation only)</DESCRIPTION>
+      <LONGDESCRIPTION>Numeric user id, only on UNIX systems</LONGDESCRIPTION>
+      <CONSTRAINT>
+        <MAYBEEMPTY>true</MAYBEEMPTY>
+      </CONSTRAINT>
+    </INPUT>
+      <INPUT>
+        <NAME>USERGROUP_USER_HOME_PERSONNALIZE</NAME>
+        <DESCRIPTION>Use the default home directory</DESCRIPTION>
+        <LONGDESCRIPTION>If not checked, it will set the defined home directory if "Policy to apply to this acocunt" if "Create/Update"</LONGDESCRIPTION>
```
Comment (amousset) [inline techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml]: account

File: techniques/systemSettings/userManagement/userManagement/8.0/userManagement.ps1.st (hunk 5 above)
Comment (Fdall) [reply techniques/systemSettings/userManagement/userManagement/8.0/userManagement.ps1.st]: it is to keep compatibility with old versions... But now we have the Audit mode to do that feature.
To remove when making a new clean version in dsc AND cf-engine.

**Comment by Fdall** [comment]
Commit modified

**Comment by amousset** [comment]
Missing entry in maintained-techniques

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

File: techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml (hunk 3 above)
Comment (peckpeck) [reply techniques/systemSettings/userManagement/userManagement/8.0/metadata.xml]: Still there

**Pull request merged**